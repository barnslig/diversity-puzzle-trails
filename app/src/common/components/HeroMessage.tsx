import { Box, Container, makeStyles, Typography } from "@material-ui/core";
import * as React from "react";
import theme from "../../app/theme";

type HeroMessageProps = {
  /**
   * Title
   */
  title: React.ReactNode;

  /**
   * Optional description text
   */
  description?: React.ReactNode;

  /**
   * Optional nodes placed after the description
   */
  after?: React.ReactNode;
};

const useStyle = makeStyles<typeof theme, HeroMessageProps>((theme) => ({
  root: {
    bottom: 0,
    left: 0,
    paddingBottom: (props) => theme.spacing(props.after ? 7 : 14),
    paddingTop: theme.spacing(10),
    position: "absolute",
    textAlign: "center",
    width: "100%",
  },
}));

/**
 * A hero message displayed on a page instead of its actual content
 * Example use cases: Game is paused, Character not set, ...
 */
const HeroMessage = (props: HeroMessageProps) => {
  const classes = useStyle(props);
  const { title, description, after } = props;

  return (
    <Box component="main" className={classes.root}>
      <Container maxWidth="sm">
        {title && (
          <Typography variant="h5" component="h2" gutterBottom>
            {title}
          </Typography>
        )}
        {description && (
          <Typography variant="body2" color="textSecondary">
            {description}
          </Typography>
        )}
        {after}
      </Container>
    </Box>
  );
};

export default HeroMessage;
