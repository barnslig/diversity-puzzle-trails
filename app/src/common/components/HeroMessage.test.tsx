import * as React from "react";
import render from "../testing/render";

import HeroMessage from "./HeroMessage";

it("renders correctly", () => {
  const { container } = render(
    <HeroMessage
      title="Test title"
      description="Test description"
      after="test 123"
    />
  );
  expect(container.firstChild).toMatchSnapshot();
});
