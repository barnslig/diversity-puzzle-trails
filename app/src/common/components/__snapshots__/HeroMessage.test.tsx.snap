// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  class="MuiBox-root css-elj48x"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
    >
      Test title
    </h2>
    <p
      class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      Test description
    </p>
    test 123
  </div>
</main>
`;
